src/app/qgsfeatureaction.cpp

```diff
@@ -314,6 +318,26 @@ void QgsFeatureAction::onFeatureSaved( const QgsFeature &feature )
   }
 }
 
+void QgsFeatureAction::hideParentWidget()
+{
+  if ( parentWidget() )
+  {
+    QWidget *dialog = parentWidget()->parentWidget();
```

Comment:
If I understand correctly, we want to hide the window of the parent widget, why do we have to get the parent of the parent ?

Reply:
Before this implementation, the a dialog was created in `QgsFeatureAction` with no parent and was set as parent of the `QgsFeatureAction` here https://github.com/qgis/QGIS/pull/47392/files#diff-030b68e25d491f6f5baf29b403af2e9f22e1f7de23a22acca5aff3d842020727R268
This means when the dialog was closed (destroyed) the `QgsFeatureAction` object got deleted as well. 

When we set now the relation editor as the parent of the `QgsFeatureAction`, it's set as the parent of the dialog on it's creation and on the mentioned line the dialog became parent of `QgsFeatureAction`. So the relation editor is the parent's (dialog) parent of `QgsFeatureAction`.

Btw. to name the variable dialog here is confusing. Since the `parentWidget()` is the dialog.
I will address that.